include/ews/ews.hpp

```diff
@@ -7602,6 +7602,545 @@ namespace ews
+            //! Standard destructor
+            ~property_tag(){};
+
+            //! Returns this property as a string.
+            const std::string& str() const EWS_NOEXCEPT { return _str; }
+
+        private:
+            std::string _str;
+        };
+
+#ifdef EWS_HAS_NON_BUGGY_TYPE_TRAITS
+        static_assert(std::is_default_constructible<property_tag>::value, "");
+        static_assert(std::is_copy_constructible<property_tag>::value, "");
+        static_assert(std::is_copy_assignable<property_tag>::value, "");
+        static_assert(std::is_move_constructible<property_tag>::value, "");
+        static_assert(std::is_move_assignable<property_tag>::value, "");
+#endif
+
+        //! \brief Represents the <tt>\<PropertyId\></tt> of an
+        //! <tt>\<ExtendedFieldUri\></tt>.
+        //!
+        //! Identifies an extended property by its dispatch ID. The dispatch ID
+        //! can be identified in either decimal or hexadecimal formats.
+        //! This property must be coupled with either DistinguishedPropertySetId
+        //! or PropertySetId. If this attribute is used, the PropertyName and
+        //! PropertyTag attributes cannot be used. This attribute is optional.
+        class property_id final
+        {
+        public:
+//! Default constructor
+#ifdef EWS_HAS_DEFAULT_AND_DELETE
+            property_id() = default;
+#else
+            property_id(){};
+#endif
+            //! Constructor to create a PropertyId property
+            explicit property_id(std::string str) : _str(std::move(str)){};
+            //! Standard destructor
+            ~property_id(){};
+
+            //! Returns this property as a string.
+            const std::string& str() const EWS_NOEXCEPT { return _str; }
+
+        private:
+            std::string _str;
+        };
+
+        //! \brief Represents the <tt>\<PropertyType\></tt> of an
+        //! <tt>\<ExtendedFieldUri\></tt>.
+        //!
+        //! Represents the property type of a property tag. This corresponds
+        //! to the least significant word in a property tag. The PropertyType
+        //! Attribute table later in this topic contains the possible values
+        //! for this attribute. This attribute is required.
+        class property_type final
+        {
+        public:
+//! Default constructor
+#ifdef EWS_HAS_DEFAULT_AND_DELETE
+            property_type() = default;
+#else
+            property_type(){};
+#endif
+            //! Constructor to create a PropertyType property
+            explicit property_type(std::string str) : _str(std::move(str)){};
+            //! Standard destructor
+            ~property_type(){};
+
+            //! Returns this property as a string.
+            const std::string& str() const EWS_NOEXCEPT { return _str; }
+
+            extended_field_uri
+            form_xml_element(const rapidxml::xml_node<>& elem);
+
+        private:
+            std::string _str;
+        };
```

Comment:
You have made `distinguished_property_set_id`, `property_set_id`, `property_tag`, `property_id`, `property_type` inner classes of `extended_field_uri`. I think this is good because these are actually not EWS types but attributes of `<ExtendedFieldURI/>`.

However, where is `property_name`? You should introduce this as a distinct type, too. Use of a `std::string` here but use of distinct types everywhere else is inconsistent.